rock-paper-scissors/service/src/main.rs

```diff
@@ -0,0 +1,223 @@
+use oasis_std::{Address, Context};
+#[macro_use]
+extern crate serde;
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq)]
+pub enum Move {
+    Null,
+    Rock,
+    Paper,
+    Scissors,
+}
+
+trait ValidMove {
```

Comment:
nice idea, but you can actually `impl` these methods directly on `Move` without using a trait.
If you want to make your syntax look like `move < other_move`, you can impl [PartialOrd](https://doc.rust-lang.org/std/cmp/trait.PartialOrd.html) for Move

Reply:
I think PartialOrd requires transitive property? RockPaperScissors has a "circular" order so that's why I didn't do it that way :O

Reply:
I guess the point still remains that maybe the `trait ValidMove` is not required, and that these methods can be implemented directly on `Move` as @nhynes suggests

Reply:
absolutely, will do it that way